ddtrace/internal/datadog/profiling/dd_wrapper/src/sample.cpp

```diff
@@ -0,0 +1,336 @@
+#include "sample.hpp"
+#include "global_cache.hpp"
+
+#include <thread>
+
+using namespace Datadog;
+
+Sample::Sample(SampleType type, unsigned int _max_nframes)
```

Comment:
does this happen under a lock?

Reply:
Fixed even better because now each caller of `start_sample` has its own opaque pointer